core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/recoverer.go

```diff
@@ -112,34 +114,40 @@ func (r *logRecoverer) Start(pctx context.Context) error {
 	r.lock.Unlock()
 
 	blockTimeResolver := newBlockTimeResolver(r.poller)
-	blockTime, err := blockTimeResolver.BlockTime(ctx, defaultSampleSize)
-	if err != nil {
-		// TODO: TBD exit or just log a warning
-		// return fmt.Errorf("failed to compute block time: %w", err)
-		r.lggr.Warnw("failed to compute block time", "err", err)
-	}
-	if blockTime > 0 {
-		r.blockTime.Store(int64(blockTime))
+	updateBlockTime := func(ctx context.Context) {
+		blockTime, err := blockTimeResolver.BlockTime(ctx, defaultSampleSize)
+		if err != nil {
+			r.lggr.Warnw("failed to compute block time", "err", err)
+			return
+		}
+		if blockTime > 0 {
+			r.blockTime.Store(int64(blockTime))
```

Comment:
can we add a debug log with the new block time here

Reply:
and also maybe a warning if it changed by a lot?

Reply:
Did I already remove the TODO? I don't see one.

Reply:
ah yes, removed from line 117, my bad